### PR TITLE
Improve trio.Path.iterdir

### LINKED ISSUE #501: trio.Path.iterdir wrapping is broken
Given `pathlib.Path.iterdir` returns a generator that does IO access on each iteration, `trio.Path.iterdir` is currently broken given it currently only generates the generator asynchronously (which I suppose is pointless given there is no need for IO at generator creation)

The solution would be to modify  `trio.Path.iterdir` to return an async generator, however this means creating a special case given the current implementation is only an async wrapper on `pathlib.Path.iterdir`.

### PR DESCRIPTION
**Pull request by touilleMan**
Closes #501

### CODE REVIEW

File: trio/_path.py [hunk 1]
```diff
@@ -128,6 +128,19 @@ def __init__(self, *args):
 
         self._wrapped = pathlib.Path(*args)
 
+    async def iterdir(self):
+        """
+        Almost (see below) like :meth:`iterdir.Path.iterdir`, but async.
+
+        Unlike :meth:`iterdir.Path.iterdir`, this method loads the entire
```
Comment (Fuyukai) [inline trio/_path.py]: Did you mean `pathlib.Path.iterdir`?

Comment (touilleMan) [reply trio/_path.py]: Good catch ^^

**Comment by njsmith** [comment]
Code looks good to me. I edited the docstring, and also edited your initial comment to say "closes #501 ", so when this is merged it will automagically close the issue.

Please go ahead and merge after confirming you like my edited docstring :-)

**Comment by touilleMan** [comment]
Your comments&docstrings are always the best anyway ;-)

**Pull request merged**